CascadeEnergy/mutton
# SEN-1025 -- Mutton test command

To test:
* Install mutton globally from this branch `npm install -g git://github.com/cberube/mutton.git#SEN-1025`
* Generate an updated mutton config file: `mutton config`
* Edit the `~/.mutton/conf.json` file to include the correct region and account data, and any other variables (see sample file below)
* Verify that deployment still works as expected -- note that the base path for lambda functions is now specified on the command line, not in the config file (e.g. `mutton deploy /var/cascade/lambda/ digest/**`
* Execute a test command: `mutton test digest-correctData /var/cascade/lambda/digest/normalDataAvailable/correctData/support/samples/events/SEN-1042-001.json` -- the first argument is the name of the lambda function to test, the second argument is a JSON file containing the event to pass to the function (this will be treated as a mustache template and any variables in the configuration file will be replaced in the event)
* Verify that mutton invokes the function, waits for log entries from the function to appear, then collects and displays those log entries

**Sample ~/.mutton/conf.json file:**

```json
{
  "mutton": {
    "deployPath": "/tmp",
    "aws": {
      "region": "us-west-2"
    },
    "variables": {
      "accountId": "123456789012",
      "region": "us-west-2"
    }
  }
}
```
